src/pages/canvasPage/Canvas.tsx

```diff
@@ -0,0 +1,197 @@
+import { useRef, useEffect, useState } from "react";
+
+import { Box, IconButton } from "@mui/material";
+import ContentCutIcon from "@mui/icons-material/ContentCut";
+import DeleteIcon from "@mui/icons-material/Delete";
+
+import { primaryColor } from "../../style";
+
+type CanvasProps = {
+  width: number;
+  height: number;
+  imageUpload: File;
+  setThumbnail: (thumbnail: Blob) => void;
+};
+
+const lineWidth = 5;
+const rectHeight = 100;
+const rectX = lineWidth / 2;
+const rectYMin = lineWidth / 2;
+
+export function Canvas(props: CanvasProps) {
+  const [isCut, setIsCut] = useState(false);
+  const [imageToCut, setImageToCut] = useState<HTMLImageElement | undefined>(
+    undefined
+  );
+  const rectY = useRef(rectYMin);
+
+  const canvasRef = useRef<HTMLCanvasElement>(null);
+
+  const rectWidth = props.width - lineWidth;
+  const rectYMax = props.height - rectYMin - rectHeight;
+
+  let isRectMoving = false;
+
+  function handleClickCut() {
+    setIsCut(isCut ? false : true);
+  }
+
+  useEffect(() => {
+    const canvas = canvasRef.current;
+    if (canvas === null) {
+      return;
+    }
+    const context = canvas.getContext("2d");
+    const cRect = canvas.getBoundingClientRect();
+    const image = new Image();
+
+    if (isCut && imageToCut !== undefined) {
+      context?.clearRect(0, 0, props.width, props.height);
+
+      if (rectY.current < rectYMin) {
+        rectY.current = rectYMin;
+      }
+      if (rectY.current > rectYMax) {
+        rectY.current = rectYMax;
+      }
+
+      let ySource =
+        (((rectY.current * 100) / props.height) * imageToCut.height) / 100;
+
+      context?.drawImage(
+        imageToCut,
+        0,
+        ySource,
+        imageToCut.width,
+        imageToCut.height / 4,
+        0,
+        0,
+        rectWidth,
+        rectHeight
+      );
+
+      const croppedImage = context?.getImageData(0, 0, rectWidth, rectHeight);
+      const croppedCanvas = document.createElement("canvas");
+
+      if (croppedCanvas !== undefined && croppedImage !== undefined) {
+        croppedCanvas.width = croppedImage?.width;
+        croppedCanvas.height = croppedImage?.height;
+
+        const contextCroppedCanvas = croppedCanvas.getContext("2d");
+
+        contextCroppedCanvas?.putImageData(croppedImage, 0, 0);
+        const dataURL = croppedCanvas.toDataURL();
```

Comment:
Diese recht komplexen Zeilen könnte man auslagern in eine Funktion, zB: canvasToBlob